src/lib/drivers/device/CDev.cpp

```diff
@@ -338,31 +396,22 @@ CDev::poll_notify(pollevent_t events)
 void
 CDev::poll_notify_one(px4_pollfd_struct_t *fds, pollevent_t events)
 {
-	DEVICE_DEBUG("CDev::poll_notify_one");
-
-#ifdef __PX4_NUTTX
-	int value = fds->sem->semcount;
```

Comment:
Any idea why this was here in the first place?

Reply:
My guess is that the original author thought it's not good to unlock the semaphore twice.

Reply:
> The next waiter did not wait when it should have. This should be investigated.

There is no next waiter, the semaphore is only used for that specific poll() call. But yes, please go through the code.

Reply:
Just to be double sure I tested that:
- the semaphore counter goes up to 2 at this location in some cases
- at the same time no `poll()` does ever return early without an event

So it's not needed.